kaldb/src/main/java/com/slack/kaldb/logstore/search/LogIndexSearcherImpl.java

```diff
@@ -69,36 +71,36 @@ public LogIndexSearcherImpl(
     this.searcherManager = searcherManager;
     this.analyzer = new StandardAnalyzer();
     this.chunkSchema = chunkSchema;
+    this.openSearchAggregationAdapter = new OpenSearchAggregationAdapter(chunkSchema);
   }
 
   // Lucene's query parsers are not thread safe. So, create a new one for every request.
   private QueryParser buildQueryParser() {
     return new KaldbQueryParser(SystemField.ALL.fieldName, analyzer, chunkSchema);
   }
 
+  @Override
   public SearchResult<LogMessage> search(
       String dataset,
       String queryStr,
       long startTimeMsEpoch,
       long endTimeMsEpoch,
       int howMany,
-      int bucketCount) {
+      AggBuilder aggBuilder) {
 
     ensureNonEmptyString(dataset, "dataset should be a non-empty string");
     ensureNonNullString(queryStr, "query should be a non-empty string");
     ensureTrue(startTimeMsEpoch >= 0, "start time should be non-negative value");
     ensureTrue(startTimeMsEpoch < endTimeMsEpoch, "end time should be greater than start time");
     ensureTrue(howMany >= 0, "hits requested should not be negative.");
-    ensureTrue(bucketCount >= 0, "bucket count should not be negative.");
-    ensureTrue(howMany > 0 || bucketCount > 0, "Hits or histogram should be requested.");
+    ensureTrue(howMany > 0 || aggBuilder != null, "Hits or aggregation should be requested.");
```

Comment:
I think we could add another check if both are passed? I think the current behaviour is if both are present we will follow the `howMany` block?

Reply:
If both are passed we build both result sets - the actual `N` log results (hits), and the aggregations. Requesting both hits and aggs is a common pattern, and is used by the custom Grafana plugin as well.

Reply:
offcourse. I should have looked more closely at the `if-else` block